vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -30,20 +29,165 @@ WindowsModalHostComponentView::WindowsModalHostComponentView(
           ComponentViewFeatures::Default & ~ComponentViewFeatures::Background,
           false) {
   m_props = std::make_shared<facebook::react::ModalHostViewProps const>();
+  m_context = reactContext; // save context
   m_visual = compContext.CreateSpriteVisual();
 }
 
+winrt::Microsoft::ReactNative::ComponentView WindowsModalHostComponentView::Create(
+    const winrt::Microsoft::ReactNative::Composition::ICompositionContext &compContext,
+    facebook::react::Tag tag,
+    winrt::Microsoft::ReactNative::ReactContext const &reactContext) noexcept {
+  return winrt::make<WindowsModalHostComponentView>(compContext, tag, reactContext);
+}
+
+// constants for creating a new windows (code mostly taken from LogBox)
+constexpr PCWSTR c_modalWindowClassName = L"MS_REACTNATIVE_MODAL";
+constexpr auto CompHostProperty = L"CompHost";
+const int MODAL_DEFAULT_WIDTH = 500;
+const int MODAL_DEFAULT_HEIGHT = 500;
+
+// creates a new modal window
+void WindowsModalHostComponentView::ShowOnUIThread() {
```

Comment:
Can you refactor this slightly into two functions, something like:

1. `EnsureModalCreated()` which does the work of checking m_hwnd and building the window
2. `ShowOnUIThread()` which calls that function then calls `ShowWindow`, etc.

I think we'll run into other places where we'll want to create the window so it's ready for UI, even if we don't show it right away.